api/internal/handler/data_loader/route_export.go

```diff
@@ -0,0 +1,441 @@
+		}
+
+		if route.RemoteAddrs != nil {
+			extensions["x-apisix-remote_addrs"] = route.RemoteAddrs
+		}
+
+		if route.FilterFunc != "" {
+			extensions["x-apisix-filter_func"] = route.FilterFunc
+		}
+
+		if route.Script != nil {
+			extensions["x-apisix-script"] = route.Script
+		}
+
+		if route.ServiceProtocol != "" {
+			extensions["x-apisix-service_protocol"] = route.ServiceProtocol
+		}
+		if route.Vars != nil {
+			extensions["x-apisix-vars"] = route.Vars
+		}
+
+		// analysis route.URIs
+		routeURIs := []string{}
+		if route.URI != "" {
+			routeURIs = append(routeURIs, route.URI)
+		}
+
+		if route.Uris != nil {
+			routeURIs = route.Uris
+		}
+
+		for _, uri := range routeURIs {
+			if strings.Contains(uri, "*") {
+				paths[strings.Split(uri, "*")[0]+"{params}"] = pathItem
+				// add params introduce
+				paramsRefs = append(paramsRefs, &openapi3.ParameterRef{
+					Value: &openapi3.Parameter{
+						In:          "path",
+						Name:        "params",
+						Required:    true,
+						Description: "params in path",
+						Schema:      &openapi3.SchemaRef{Value: &openapi3.Schema{Type: "string"}}}})
+			} else {
+				paths[uri] = pathItem
+			}
+		}
+
+		//Parse Route Plugins
+		path, secSchemas, paramsRefs, plugins, err = parseRoutePlugins(route, paramsRefs, plugins, path, servicePlugins, secSchemas, requestBody)
+		if err != nil {
+			log.Errorf("parseRoutePlugins err: ", err)
+			return nil, err
+		}
+		if plugins != nil {
+			extensions["x-apisix-plugins"] = plugins
+		}
+
+		extensions["x-apisix-priority"] = route.Priority
+		extensions["x-apisix-status"] = route.Status
+		extensions["x-apisix-enable_websocket"] = route.EnableWebsocket
+		path.Extensions = extensions
+		path.Parameters = paramsRefs
+		path.RequestBody = &openapi3.RequestBodyRef{Value: requestBody}
+		path.Responses = openapi3.NewResponses()
+
+		for i := range route.Methods {
+			switch strings.ToUpper(route.Methods[i]) {
+			case http.MethodGet:
+				//pathItem.Get = path
+				pathItem.Get = parsePathItem(path, http.MethodGet)
+			case http.MethodPost:
+				//pathItem.Post = path
+				pathItem.Post = parsePathItem(path, http.MethodPost)
+			case http.MethodPut:
+				//pathItem.Put = path
+				pathItem.Put = parsePathItem(path, http.MethodPut)
+			case http.MethodDelete:
+				//pathItem.Delete = path
+				pathItem.Delete = parsePathItem(path, http.MethodDelete)
+			case http.MethodPatch:
+				//pathItem.Patch = path
+				pathItem.Patch = parsePathItem(path, http.MethodPatch)
+			case http.MethodHead:
+				//pathItem.Head = path
+				pathItem.Head = parsePathItem(path, http.MethodHead)
+			}
+		}
+	}
+
+	components.SecuritySchemes = secSchemas
+	swagger := openapi3.Swagger{
+		OpenAPI:    openApi,
+		Info:       &openapi3.Info{Title: title, Version: openApi},
+		Paths:      paths,
+		Components: *components,
+	}
+	return &swagger, nil
+}
+
+func parseLabels(route *entity.Route, serviceLabels map[string]string, labels map[string]string) (map[string]string, error) {
```

Comment:
What's the label args use for?

Reply:
When service and route have labels at the same time, use route's label. When route has no label, service sometimes uses service's label. This function is used to process this logic

Reply:
No, I didn't see the lables args was used.

Reply:
I don't quite understand your question. Do you mean you don't see users or business scenarios using this field? I don't know if my understanding is correct, because more coverage possibilities are considered everywhere to ensure that the data is not lost. After discussing with nic-chen, this processing function is added.